HelpSource/Classes/ArrayedCollection.schelp

```diff
@@ -393,7 +393,7 @@ code::
 ::
 
 method::collect
-Answer a new collection which consists of the results of function evaluated for each item in the collection. The function is passed two arguments, the item and an integer index. See link::Classes/Collection:: helpfile for examples.
+Return a new collection which consists of the results of function evaluated for each item in the collection. The function is passed two arguments, the item and an integer index. See link::Classes/Collection:: helpfile for examples.
```

Comment:
All the changes should all be return**s**. 

If it's singular it reads like an instruction to the implementor that the method must return something, or as an instructor to the caller that they must do something.

Reply:
At first glance I thought I would have to change them all to 'Returns'. Then I thought that the first sentence of the method description might be an imperative sentence because most of the first words are 'Answer'. So I changed it to 'Return'. However, I think 'Returns' sounds better. I can change them all to 'Returns'.

Reply:
I suppose it's a question of who is the documentation for, the user or the implementer? I err on the side of the user here.

Reply:
I agree with @JordanHendersonMusic. However, I think changing 'Return' to 'Returns' is outside the scope of this PR. I should open a new PR after this one is merged:
- [method::(\w*)\nReturn ]: 144 results in 29 files
- [method::(\w*)\nReturns ]: 183 results in 43 files

### [method::(\w*)\nReturn ]: 144 results in 29 files
```
HelpSource/Classes/Array.schelp:
  144  
  145: method::mirror
  146: Return a new Array which is the receiver made into a palindrome.
  147  The receiver is unchanged.

  151  
  152: method::mirror1
  153: Return a new Array which is the receiver made into a palindrome with the last element removed.
  154  This is useful if the list will be repeated cyclically, the first element will not get played twice.

  160  
  161: method::mirror2
  162: Return a new Array which is the receiver concatenated with a reversal of itself.
  163  The center element is duplicated. The receiver is unchanged.

  176  
  177: method::dupEach
  178: Return a new Array whose elements are repeated n times. The receiver is unchanged.
  179  code::

  184  
  185: method::rotate
  186: Return a new Array whose elements are in rotated order. The receiver is unchanged.
  187  code::

  195  
  196: method::pyramid
  197: Return a new Array whose elements have been reordered via one of 10 "counting" algorithms.
  198  Run the examples to see the algorithms.

  214  
  215: method::sputter
  216: Return a new Array of length maxlen with the items partly repeated (random choice of given probability).
  217  code::

  276  
  277: method::slide
  278: Return a new Array whose elements are repeated subsequences from the receiver.
  279  Easier to demonstrate than explain.

HelpSource/Classes/Array2D.schelp:
  70  
  71: method::asArray
  72: Return a flat array containing the elements.
  73  code::

HelpSource/Classes/ArrayedCollection.schelp:
   45  
   46: method::size
   47: Return the number of elements the ArrayedCollection.
   48  
   49: method::maxSize
   50: Return the maximum number of elements the ArrayedCollection can hold. For example, link::Classes/Array::s may initialise themselves with a larger capacity than the number of elements added.
   51  code::

   68  
   69: method::at
   70: Return the item at strong::index::.
   71  

  132  
  133: method::indexOf
  134: Return the first index containing an item which matches strong::item::. Elements are checked for identity (not for equality).
  135  code::

  139  
  140: method::includes
  141: Return a boolean indicating whether the collection contains anything matching strong::item::. Elements are checked for identity (not for equality).
  142  code::

  147  
  148: method::indexOfGreaterThan
  149: Return the first index containing an item which is greater than strong::item::.
  150  code::

  299  
  300: method::copyRange
  301: Return a new ArrayedCollection which is a copy of the indexed slots of the receiver from strong::start:: to strong::end::. If strong::end:: < strong::start::, an empty ArrayedCollection is returned.
  302  code::

  312  
  313: method::copySeries
  314: Return a new ArrayedCollection consisting of the values starting at strong::first::, then every step of the distance between strong::first:: and strong::second::, up until strong::last::. If strong::second:: is code::nil::, a step of 1 or -1 is used as appropriate.
  315  

  370  
  371: method::reverse
  372: Return a new ArrayedCollection whose elements are reversed.
  373  code::

  394  
  395: method::collect
  396: Return a new collection which consists of the results of function evaluated for each item in the collection. The function is passed two arguments, the item and an integer index. See link::Classes/Collection:: helpfile for examples.
  397  

  499  
  500: method::replace
  501: Return a new array in which a number of elements have been replaced by another. Elements are checked for equality (not for identity).
  502  code::

  512  
  513: method::asRandomTable
  514: Return an integral table that can be used to generate random numbers with a specified distribution.
  515  (see link::Guides/Randomness:: helpfile for a more detailed example)

  532  
  533: method::msgSize
  534: Return the size of an osc message in bytes
  535  code::

  539  
  540: method::bundleSize
  541: Return the size of an osc bundle in bytes
  542  code::

HelpSource/Classes/BusPlug.schelp:
  45  
  46: method::server
  47: Return the server that the BusPlug runs on.
  48  

HelpSource/Classes/Collection.schelp:
   74  
   75: method::size
   76: Return the number of objects contained in the Collection.
   77  code::

  150  
  151: method::atAll
  152: Return a collection of all the items for the keys.
  153  code::

  200  
  201: method::collect
  202: Return a new collection which consists of the results of strong::function:: evaluated for each item in the collection. The function is passed two arguments, the item and an integer index.
  203  code::

  207  
  208: method::select
  209: Return a new collection which consists of all items in the receiver for which strong::function:: answers link::Classes/True::. The function is passed two arguments, the item and an integer index.
  210  code::

  214  
  215: method::reject
  216: Return a new collection which consists of all items in the receiver for which strong::function:: answers link::Classes/False::. The function is passed two arguments, the item and an integer index.
  217  code::

  221  
  222: method::detect
  223: Return the first item in the receiver for which strong::function:: answers link::Classes/True::. The function is passed two arguments, the item and an integer index.
  224  code::

  293  
  294: method::count
  295: Return the number of items for which strong::function:: answers link::Classes/True::. The function is passed two arguments, the item and an integer index.
  296  code::

  299  
  300: method::occurrencesOf
  301: Return the number of items in the receiver which are equal to anObject.
  302  code::

  305  
  306: method::sum
  307: Return the sum of the results of strong::function:: evaluated for each item in the receiver. The function is passed two arguments, the item and an integer index.
  308  code::

  312  
  313: method::maxItem
  314: Return the maximum of the results of strong::function:: evaluated for each item in the receiver. The function is passed two arguments, the item and an integer index.
  315  If function is nil, then answer the maximum of all items in the receiver.

  319  
  320: method::minItem
  321: Return the minimum of the results of strong::function:: evaluated for each item in the receiver. The function is passed two arguments, the item and an integer index.
  322  If function is nil, then answer the minimum of all items in the receiver.

  326  
  327: method::maxIndex
  328: Return the index of the maximum of the results of strong::function:: evaluated for each item in the receiver. The function is passed two arguments, the item and an integer index.
  329  If function is nil, then answer the maximum of all items in the receiver.

  334  
  335: method::minIndex
  336: Return the index of the minimum of the results of strong::function:: evaluated for each item in the receiver. The function is passed two arguments, the item and an integer index.
  337  If function is nil, then answer the minimum of all items in the receiver.

  379  
  380: method::asBag
  381: Return a link::Classes/Bag:: to which all items in the receiver have been added.
  382  code::

  385  
  386: method::asList
  387: Return a link::Classes/List:: to which all items in the receiver have been added.
  388  code::

  391  
  392: method::asSet
  393: Return a link::Classes/Set:: to which all items in the receiver have been added.
  394  code::

  397  
  398: method::asSortedList
  399: Return a link::Classes/SortedList:: to which all items in the receiver have been added.
  400  code::

  403  
  404: method::asDict
  405: Return a corresponding dictionary. This is part of the link::Reference/Key-Value-Pairs:: interface.
  406  argument::mergeFunc

  410  
  411: method::asAssociations
  412: Return an array of link::Classes/Association::s. If the first item of the list is already an associiation, return itself.  This is part of the link::Reference/Key-Value-Pairs:: interface.
  413  argument::class

  415  
  416: method::asPairs
  417: Return an array with alternating key value pairs, like code::[\freq, 1848, \amp, 0.2]::. This is part of the link::Reference/Key-Value-Pairs:: interface.
  418  argument::class

  421  
  422: method::asEvent
  423: Return an link::Classes/Event::: with the key value pairs. See link::Reference/Key-Value-Pairs::.
  424  

  517  
  518: method::sect
  519: Return the set theoretical intersection of this and strong::that::.
  520  code::

  524  
  525: method::union
  526: Return the set theoretical union of this and strong::that::.
  527  code::

  531  
  532: method::difference
  533: Return the set of all items which are elements of this, but not of strong::that::.
  534  code::

  538  
  539: method::symmetricDifference
  540: Return the set of all items which are not elements of both  this and strong::that::.
  541  this -- that

HelpSource/Classes/ContiguousBlockAllocator.schelp:
  20  
  21: method::alloc
  22: Return the starting index of a free block that is emphasis::n:: slots wide. The default is 1 slot.
  23  

HelpSource/Classes/Dictionary.schelp:
  114  
  115: method::keys
  116: Return a link::Classes/Set:: of all keys.
  117  code::

  121  
  122: method::values
  123: Return a link::Classes/List:: of all values.
  124  code::

  128  
  129: method::atAll
  130: Return an link::Classes/Array:: of all values for the given keys.
  131  code::

  135  
  136: method::getPairs
  137: Return an link::Classes/Array:: with all keys and values pairwise.
  138  

  279  
  280: method::invert
  281: Return a new dictionary with all the values as keys and vice versa.
  282  code::

  288  
  289: method::order
  290: Return an array of keys which corresponds to the order of the values of the dictionary.
  291  code::

  296  
  297: method::powerset
  298: Return the set of all subsets: here an array of all sub-dictionaries.
  299  code::

  339  
  340: method::asSortedArray
  341: Return the values in a sorted array of key value pairs. Sorted by key.
  342  code::

  353  
  354: method::asPairs
  355: Return the values in an array of alternating key value pairs, like code::[\freq, 1848, \amp, 0.2]::.  This is part of the link::Reference/Key-Value-Pairs:: interface.
  356  argument::class

HelpSource/Classes/Interval.schelp:
  27  
  28: method::size
  29: Return the number of items in the interval.
  30  code::

  33  
  34: method::at
  35: Return the indexed item in the interval.
  36  code::

HelpSource/Classes/LinkedList.schelp:
  26  
  27: method::first
  28: Return the first item in the list.
  29  
  30: method::last
  31: Return the last item in the list.
  32  
  33: method::at
  34: Return the item at the given index in the list.
  35  This requires a scan of the list and so is O(n).

HelpSource/Classes/List.schelp:
   52  
   53: method::at
   54: Return the item at strong::index::.
   55  code::

  146  
  147: method::copyRange
  148: Return a new List which is a copy of the indexed slots of the receiver from start to end.
  149  code::

  158  
  159: method::copySeries
  160: Return a new List consisting of the values starting at strong::first::, then every step of the distance between strong::first:: and strong::second::, up until strong::last::.
  161  code::

  180  
  181: method::reverse
  182: Return a new List whose elements are reversed.
  183  code::

  198  
  199: method::mirror
  200: Return a new List which is the receiver made into a palindrome. The receiver is unchanged.
  201  code::

  204  
  205: method::mirror1
  206: Return a new List which is the receiver made into a palindrome with the last element removed. This is useful if the list will be repeated cyclically, the first element will not get played twice. The receiver is unchanged.
  207  code::

  210  
  211: method::mirror2
  212: Return a new List which is the receiver concatenated with a reversal of itself. The center element is duplicated. The receiver is unchanged.
  213  code::

  216  
  217: method::stutter
  218: Return a new List whose elements are repeated strong::n:: times. The receiver is unchanged.
  219  note::It is recommended to use code::dupEach:: instead. This method is retained for backwards compatibility.::

  223  
  224: method::dupEach
  225: Return a new List whose elements are repeated strong::n:: times. The receiver is unchanged.
  226  code::

  229  
  230: method::rotate
  231: Return a new List whose elements are in rotated order. Negative strong::n:: values rotate left, positive strong::n:: values rotate right. The receiver is unchanged.
  232  code::

  237  
  238: method::pyramid
  239: Return a new List whose elements have been reordered via one of 10 "counting" algorithms. The algorithms are numbered 1 through 10. Run the examples to see the algorithms.
  240  code::

  291  
  292: method::slide
  293: Return a new List whose elements are repeated subsequences from the receiver. Easier to demonstrate than explain.
  294  code::

HelpSource/Classes/Monitor.schelp:
  180  
  181: method::group
  182: Return the group in which all mapping synths are running.
  183  
  184: method::numChannels
  185: Return the number of input channels.
  186  
  187: method::copy
  188: Return a copy of the receiver, with the same channel setting, but not running. You can run it with the settings by sending it the link::#-play:: message, and pass in any modifications you want to make.
  189  

HelpSource/Classes/Ndef.schelp:
  42  
  43: method::new
  44: Return a new node proxy and store it in a global ProxySpace under the key. If there is already an Ndef there, replace its object with the new one. The object can be any supported class, see link::Classes/NodeProxy#Supported sources:: help. Internally, all instances are kept in a link::Classes/ProxySpace::. The link::Classes/EnvironmentRedirect#dispatch:: function is called when the source is set.
  45  

  72  
  73: method::all
  74: Return the dictionary of all servers, pointing to proxyspaces with Ndefs for each.
  75  

  79  
  80: method::dictFor
  81: Return the proxyspace for a given server.
  82  

  88  
  89: method::proxyspace
  90: Return the proxyspace for this instance
  91  

HelpSource/Classes/NdefMixerOld.schelp:
  13  
  14: method::new
  15: Return a new window for a given server.
  16  

  37  
  38: method::proxyspace
  39: Return the proxyspace.
  40  

HelpSource/Classes/Nil.schelp:
  25  
  26: method::isNil
  27: Return true because this is nil. In class link::Classes/Object:: this message is defined to answer false.
  28  code::

  31  
  32: method::notNil
  33: Return false. In class link::Classes/Object:: this message answers true.
  34  code::

HelpSource/Classes/NodeProxy.schelp:
  50  
  51: method::new
  52: Return a new instance of NodeProxy.
  53  

HelpSource/Classes/Order.schelp:
  33  
  34: method::pos
  35: Return the current write position.
  36  

HelpSource/Classes/Pair.schelp:
  14  
  15: method::new
  16: Return new instance.
  17  

  24  
  25: method::size
  26: Return the size when linking across.
  27  
  28: method::depth
  29: Return the size when linking down.
  30  

HelpSource/Classes/Pattern.schelp:
  107  
  108: method::asStream
  109: Return a link::Classes/Stream:: from this pattern. One pattern can be used to produce any number of independent streams.
  110  

HelpSource/Classes/PatternProxy.schelp:
  28  
  29: method::asStream
  30: Return a link::Classes/Stream:: from this pattern. One pattern proxy can be used to produce any number of independent streams.
  31  

HelpSource/Classes/Pdef.schelp:
  180  
  181: method::player
  182: Return the current player (if the Pdef is simply used in other streams this is nil)
  183  

HelpSource/Classes/Point.schelp:
  60  
  61: method::rho
  62: Return the polar coordinate radius of the receiver.
  63  
  64: method::theta
  65: Return the polar coordinate angle of the receiver.
  66  
  67: method::dist
  68: Return the distance from the receiver to aPoint.
  69  
  70: method::transpose
  71: Return a Point whose x and y coordinates are swapped.
  72  

  86  
  87: method::asString
  88: Return a string representing the receiver.

HelpSource/Classes/PriorityQueue.schelp:
  24  
  25: method::isEmpty
  26: Return a link::Classes/Boolean:: whether the queue is empty.
  27  
  28: method::notEmpty
  29: Return a link::Classes/Boolean:: whether the queue is not empty.
  30  

HelpSource/Classes/ProxySpace.schelp:
  124  
  125: method::at
  126: Return the proxy source object at that key.
  127  

HelpSource/Classes/Rect.schelp:
   6  
   7: method::new
   8: Return a new Rect with the given upper left corner and dimensions.
   9  
  10: method::newSides
  11: Return a new Rect with the given boundaries.
  12  
  13: method::fromPoints
  14: Return a new Rect defined by the given Points.
  15  

  41  
  42: method::origin
  43: Return the upper left corner as a Point.
  44  
  45: method::extent
  46: Return a Point whose x value is the height and whose y value is the width.
  47  
  48: method::leftTop
  49: Return the upper left corner as a Point.
  50  
  51: method::rightTop
  52: Return the upper right corner as a Point.
  53  
  54: method::leftBottom
  55: Return the lower left corner as a Point.
  56  
  57: method::rightBottom
  58: Return the lower right corner as a Point.
  59  

HelpSource/Classes/SequenceableCollection.schelp:
   91  
   92: method::first
   93: Return the first element of the collection.
   94  code::

   97  
   98: method::last
   99: Return the last element of the collection.
  100  code::

  110  
  111: method::indexOf
  112: Return the index of an strong::item:: in the collection, or nil if not found. Elements are checked for identity (not for equality).
  113  code::

  117  
  118: method::indexOfEqual
  119: Return the index of something in the collection that equals the strong::item::, or nil if not found.
  120  code::

  123  
  124: method::indicesOfEqual
  125: Return an array of indices of things in the collection that equal the strong::item::, or nil if not found.
  126  code::

  131  
  132: method::indexOfGreaterThan
  133: Return the first index containing an strong::item:: which is greater than strong::item::.
  134  code::

  138  
  139: method::selectIndices
  140: Return a new collection of same type as receiver which consists of all indices of those elements of the receiver for which function answers code::true::. The function is passed two arguments, the item and an integer index.
  141  

  147  
  148: method::selectIndicesAs
  149: Return a new collection of type emphasis::class:: which consists of all indices of those elements of the receiver for which function answers code::true::. The function is passed two arguments, the item and an integer index.
  150  

  155  
  156: method::rejectIndices
  157: Return a new collection of same type as receiver which consists of all indices of those elements of the receiver for which function answers code::false::. The function is passed two arguments, the item and an integer index.
  158  

  164  
  165: method::rejectIndicesAs
  166: Return a new collection of type emphasis::class:: which consists of all indices of those elements of the receiver for which function answers code::false::. The function is passed two arguments, the item and an integer index.
  167  

  210  
  211: method::copyRange
  212: Return a new SequenceableCollection which is a copy of the indexed slots of the receiver from strong::start:: to strong::end::. If strong::end:: < strong::start::, an empty collection is returned.
  213  code::

  223  
  224: method::copyToEnd
  225: Return a new SequenceableCollection which is a copy of the indexed slots of the receiver from strong::start:: to the end of the collection.
  226  code::x.copyToEnd(a):: can also be written as code::x[a..]::
  227  
  228: method::copyFromStart
  229: Return a new SequenceableCollection which is a copy of the indexed slots of the receiver from the start of the collection to strong::end::.
  230  code::x.copyFromStart(a):: can also be written as code::x[..a]::

  539  
  540: method::order
  541: Return an array of indices that would sort the collection into order. strong::function:: is treated the same way as for the link::#-sort:: method.
  542  code::

HelpSource/Classes/Signal.schelp:
  232  
  233: method::peak
  234: Return the peak absolute value of a Signal.
  235  code::

  273  
  274: method::integral
  275: Return the integral of a signal.
  276  code::

HelpSource/Classes/String.schelp:
  234  
  235: method::asSymbol
  236: Return a link::Classes/Symbol:: derived from the String.
  237  code::

  244  
  245: method::asInteger
  246: Return an link::Classes/Integer:: derived from the String. Strings beginning with non-numeric characters return 0.
  247  code::

  250  
  251: method::asFloat
  252: Return a link::Classes/Float:: derived from the String. Strings beginning with non-numeric characters return 0.
  253  code::

  256  
  257: method::asSecs
  258: Return a link::Classes/Float:: based on converting a time string in format code::(ddd:)hh:mm:ss(.z)::, where code::z:: is any sequence of digits. This is the inverse method to link::Classes/SimpleNumber#-asTimeString::.
  259  code::

  553  
  554: method::quote
  555: Return this string enclosed in double-quote (teletype::"::) characters.
  556  code::

  559  
  560: method::zeroPad
  561: Return this string enclosed in space characters.
  562  code::

  565  
  566: method::underlined
  567: Return this string followed by dashes in the next line (teletype::-::).
  568  code::

  592  
  593: method::toUpper
  594: Return this string with uppercase letters.
  595  code::

  598  
  599: method::toLower
  600: Return this string with lowercase letters.
  601  code::

  802  
  803: method::shellQuote
  804: Return a new string suitable for use as a filename in a shell command, by enclosing it in single quotes (teletype::'::).
  805  If the string contains any single quotes they will be escaped.

  816  method::absolutePath
  817: method::asAbsolutePath
  818: Return this path as an absolute path by prefixing it with link::Classes/File#*getcwd:: if necessary.
  819  
  820: method::asRelativePath
  821: Return this path as relative to the specified path.
  822  argument::relativeTo

  830  
  831: method::basename
  832: Return the filename from a Unix path.
  833  code::

  836  
  837: method::dirname
  838: Return the directory name from a Unix path.
  839  code::

HelpSource/Classes/Tdef.schelp:
  140  
  141: method::player
  142: Return the current player (if the Tdef is simply used in other streams this is nil)
  143  

HelpSource/Classes/UnixFILE.schelp:
  12  
  13: method::pos
  14: Return the current file position
  15  
```
[method::(\w*)\nReturns ]: 183 results in 43 files
```
HelpSource/Classes/AbstractFunction.schelp:
   74  ::
   75: method::sign
   76: Returns a function that returns -1 if receiver returns a negative number, 1 if positive, and 0 if zero.
   77  code::

   98  ::
   99: method::exp
  100: Returns e to the power of this.
  101  code::

  428  
  429: method::excess
  430: Returns the difference of the receiver and its clipped form.
  431  method::<!

HelpSource/Classes/Array.schelp:
  129  
  130: method::reverse
  131: Returns a new Array whose elements are reversed. The receiver is unchanged.
  132  code::

  138  
  139: method::scramble
  140: Returns a new Array whose elements have been scrambled. The receiver is unchanged.
  141  code::

  226  
  227: method::lace
  228: Returns a new Array whose elements are interlaced sequences of the elements of the receiver's subcollections, up to size length. The receiver is unchanged.
  229  code::

  235  
  236: method::permute
  237: Returns a new Array whose elements are the nthPermutation of the elements of the receiver. The receiver is unchanged.
  238  code::

  242  
  243: method::allTuples
  244: Returns a new Array whose elements contain all possible combinations of the receiver's subcollections.
  245  code::

  249  
  250: method::wrapExtend
  251: Returns a new Array whose elements are repeated sequences of the receiver, up to size length. The receiver is unchanged.
  252  code::

  293  
  294: method::containsSeqColl
  295: Returns true if the receiver Array contains any instance of SequenceableCollection
  296  code::

  300  
  301: method::powerset
  302: Returns all possible combinations of the array's elements.
  303  code::

  400  
  401: method::asString
  402: Returns a string representing the Array. May not be compilable due to elision (...) of excessive arguments.
  403  
  404: method::asCompileString
  405: Returns a string that will compile to return an Array equal to the receiver.
  406  
  407: method::isValidUGenInput
  408: Returns true. Arrays are valid UGen inputs.
  409  
  410: method::asRawOSC
  411: Returns the OSC message as an Int8Array. Receiver can be a message or a bundle. See also link::Classes/Int8Array#-parseOSC::.
  412  code::

HelpSource/Classes/ArrayedCollection.schelp:
   54  
   55: method::isRectangular
   56: Returns true if all nested sub arrays are the same size and all elements have the same depth.
   57  This is a requirement of several nested array algorithms and formats, notably multichannel audio files.

  416  
  417: method::normalizeSum
  418: Returns the Array resulting from :
  419  code::

  428  
  429: method::normalize
  430: Returns a new Array with the receiver items normalized between strong::min:: and strong::max::.
  431  code::

  435  
  436: method::perfectShuffle
  437: Returns a copy of the receiver with its items split into two equal halves, then reconstructed by interleaving the two halves. note::the size of the collection should be even, otherwise the item directly in the middle of the collection will be lost in the shuffle.::
  438  code::

  454  
  455: method::rank
  456: Returns the number of dimensions of the collection.
  457  code::rank:: inspects the size of the left-most elements of sub-arrays only,

  470  
  471: method::shape
  472: Returns an array describing the dimension of each nested array.
  473  Requires link::#-isRectangular:: as a precondition.

  522  
  523: method::tableRand
  524: Returns a new random number from a random table.
  525  code::

HelpSource/Classes/Bag.schelp:
  20  
  21: method::contents
  22: Returns the dictionary that stores the objects in pairs (obj -> numberOfObjects)
  23  code::

HelpSource/Classes/BusPlug.schelp:
   91  
   92: method::asUGenInput
   93: Returns the appropriate output to read from the BusPlug bus (an link::Classes/InBus:: UGen)
   94  code::

   99  
  100: method::embedInStream
  101: Returns the map argument for the bus, if the bus has multiple channels, it will return an array of map args.
  102  code::

  109  
  110: method::asControlInput
  111: Returns the map argument for the bus, just like link::#-embedInStream::
  112  

  114  
  115: method::isPlaying
  116: Returns true if server is running and bus not nil. link::Classes/NodeProxy:: this returns true if the group is playing.
  117  
  118: method::isMonitoring
  119: Returns true if monitor is playing
  120  

  195  
  196: method::isNeutral
  197: Returns true if no bus has been initialized so far.
  198  

HelpSource/Classes/Collection.schelp:
  186  
  187: method::matchItem
  188: Returns link::Classes/True:: if this includes the strong::item::.
  189  

  342  
  343: method::maxSizeAtDepth
  344: Returns the maximum size of all subcollections at a certain depth (dimension)
  345  

  357  
  358: method::maxDepth
  359: Returns the maximum depth of all subcollections.
  360  

  367  
  368: method::iter
  369: Returns a link::Classes/Routine:: that returns the elements one by one.
  370  code::

  429  
  430: method::powerset
  431: Returns all possible combinations of the collection's elements.
  432  code::

  456  
  457: method::histo
  458: Returns a histogram of the collection by counting the number of values that fall into each of the strong::steps:: subdivisions (default: 100) between strong::min:: and strong::max::.
  459  If not provided, strong::min:: and strong::max:: default to the smallest and largest value in the collection, respectively.

  546  
  547: method::isSubsetOf
  548: Returns link::Classes/True:: if all elements of this are also elements of strong::that::
  549  code::

HelpSource/Classes/ControlSpec.schelp:
  62  
  63: method::constrain
  64: Returns code::value.asFloat.clip(clipLo, clipHi).round(step)::.
  65  
  66: method::range
  67: Returns code::maxval - minval::.
  68  

  71  
  72: method::gridClass
  73: Returns the link::Classes/AbstractGridLines:: subclass corresponding to the current spec, in particular its warp behavior.
  74  

HelpSource/Classes/Date.schelp:
  156  
  157: method::asString
  158: Returns asctime.
  159  

HelpSource/Classes/Dictionary.schelp:
  194  
  195: method::trueAt
  196: Returns link::Classes/True:: if the item's teletype::booleanValue:: at the key is teletype::true::, otherwise teletype::false::.
  197  

  213  
  214: method::falseAt
  215: Returns link::Classes/True:: if the item's teletype::booleanValue:: at the key is teletype::false::, otherwise teletype::true::.
  216  See link::#-trueAt:: for examples.

  220  
  221: method::includes
  222: Returns true if the specified item is stored in the Dictionary as a value. Element is checked for equality (not for identity). For identity matching see subclasses: link::Classes/IdentityDictionary:: or link::Classes/Event::.
  223  code::

  227  
  228: method::includesKey
  229: Returns true if the specified item is stored in the Dictionary as a key. Element is checked for equality (not for identity). For identity matching see subclasses: link::Classes/IdentityDictionary:: or link::Classes/Event::.
  230  code::

HelpSource/Classes/Env.schelp:
  647  
  648: method::delay
  649: Returns a new Env based on the receiver in which the start value will be held for strong::delay:: number of seconds.
  650  

  759  
  760: method::asSignal
  761: Returns a link::Classes/Signal:: of size strong::length:: created by sampling this Env at strong::length:: number of intervals. If the envelope has multiple channels (see link::#Multichannel expansion::), this method returns an array of signals.
  762  

  768  
  769: method::isSustained
  770: Returns true if this is a sustaining envelope, false otherwise.
  771  

  792  
  793: method::at
  794: Returns the value of the Env at strong::time::. If the envelope has multiple channels, this method returns an array of levels.
  795  

  899  
  900: method::streamArg
  901: Returns the same output as code::asStream::.
  902  

HelpSource/Classes/EnvGate.schelp:
  10  
  11: method::new
  12: Returns an link::Classes/EnvGen::.
  13  

HelpSource/Classes/Event.schelp:
   46  
   47: method::default
   48: Returns an empty event with link::#defaultParentEvent:: as parent.
   49  
   50: method::silent
   51: Returns an event that describes a pause of strong::dur:: duration.
   52  

  171  
  172: method::delta
  173: Returns the inter onset time - the time delay until the next event in a sequence. This usually depends on strong::\dur:: and strong::\stretch::, but may be overridden by specifying strong::\delta:: directly.
  174  code::

  188  
  189: method::isRest
  190: Returns strong::true:: if the event will be played as a rest, and strong::false:: otherwise. See link::Classes/Rest:: for a more complete discussion of rests in event patterns.
  191  

HelpSource/Classes/File.schelp:
  190  
  191: method::length
  192: Returns the current file size in bytes.
  193  

HelpSource/Classes/FreqScope.schelp:
  65  
  66: method::window
  67: Returns the window in which the link::Classes/FreqScopeView:: is placed.
  68  

HelpSource/Classes/Harmonics.schelp:
  77  
  78: method::cutoff
  79: Returns 1.0 to the nth place, fills the rest with 0.0
  80  code::

HelpSource/Classes/IdentityBag.schelp:
  30  
  31: method::contents
  32: Returns the dictionary that stores the objects in pairs (obj -> numberOfObjects)
  33  code::

HelpSource/Classes/IdentityDictionary.schelp:
  110  
  111: method::includesKey
  112: Returns true if the key exists in the dictionary
  113  code::

  116  
  117: method::findKeyForValue
  118: Returns the key for a given value (it'll return the first it finds, so this may be ambiguous).
  119  code::

HelpSource/Classes/IdentitySet.schelp:
  46  
  47: method::findMatch
  48: Returns item if it is in the collection, otherwise returns nil.
  49  code::

HelpSource/Classes/LazyEnvir.schelp:
  27  
  28: method::at
  29: Returns a reference to the object at key.
  30  

  96  
  97: method::makeProxy
  98: Returns a new placeholder object. This is used internally and can be overridden to implement other lazy environments.

HelpSource/Classes/List.schelp:
   36  
   37: method::asArray
   38: Returns a new link::Classes/Array:: based upon this List.
   39  
   40: method::array
   41: Returns the List's Array, allowing it to be manipulated directly. This should only be necessary for exotic manipulations not implemented in List or its superclasses.
   42  code::

  192  
  193: method::scramble
  194: Returns a new List whose elements have been scrambled. The receiver is unchanged.
  195  code::

  249  
  250: method::lace
  251: Returns a new List whose elements are interlaced sequences of the elements of the receiver's subcollections, up to size strong::length::. The receiver is unchanged.
  252  code::

  260  
  261: method::permute
  262: Returns a new List whose elements are the strong::nthPermutation:: of the elements of the receiver. The receiver is unchanged.
  263  code::

  269  
  270: method::wrapExtend
  271: Returns a new List whose elements are repeated sequences of the receiver, up to size strong::length::. The receiver is unchanged.
  272  code::

HelpSource/Classes/Monitor.schelp:
  177  
  178: method::isPlaying
  179: Returns true if the group is still playing.
  180  

HelpSource/Classes/Nil.schelp:
   51  
   52: method::booleanValue
   53: Returns false.
   54  code::

   59  
   60: method::rate
   61: Returns nil.
   62  
   63: method::numChannels
   64: Returns nil.
   65  
   66: method::isPlaying
   67: Returns false.
   68  
   69: method::dependants
   70: Returns an empty IdentitySet.
   71  
   72: method::awake
   73: Returns nil.
   74  
   75: method::nextTimeOnGrid
   76: Returns clock.nextTimeOnGrid.
   77  
   78: method::asQuant
   79: Returns Quant.default.
   80  
   81: method::matchItem
   82: Returns true.
   83  

   93  
   94: method::asCollection
   95: Returns empty array.
   96  
   97: method::get
   98: Returns prevVal.
   99  
  100  
  101: method::asSpec
  102: Returns the default ControlSpec
  103  

  109  
  110: method::appendStream
  111: Returns stream.
  112  

  125  
  126: method::add
  127: Returns an array with the value. This makes it unnecessary to initialize when adding to a variable.
  128  code::

  133  
  134: method::addAll
  135: Returns an array with all the values. This makes it unnecessary to initialize when adding to a variable.
  136  code::

  159  
  160: method::addFunc
  161: Returns a function or a FunctionList.
  162  This method is used to add multiple functions to already existing ones.

HelpSource/Classes/NodeProxy.schelp:
  381  
  382: method::sources
  383: Returns an array of all sources
  384  
  385: method::source
  386: Returns the first source.
  387  

  420  
  421: method::loaded
  422: Returns true if the object has been initialized on the server, e.g. a synthDef has been stored.
  423  
  424: method::paused
  425: Returns true if the processes are paused.
  426  

  527  
  528: method::controlNames
  529: Returns the link::Classes/ControlName:: objects of all slots, strong::except:: the names of this list (default: code::[\out, \i_out, \gate, \fadeTime]::, which are used internally). When the node proxy received a code::set:: for a given key, the code::defaultValue:: of the corresponding code::ControlName:: contains the object that was passed, which might be any object whatsoever.
  530  
  531: method::controlKeys
  532: Returns the keys (symbols) of all control names objects of all slots, strong::except:: the names of this list. (default: none).
  533  

  614  
  615: method::edit
  616: Returns a new instance of link::Classes/NodeProxyEditor:: for this proxy.
  617  

HelpSource/Classes/NodeProxyEditor.schelp:
  36  
  37: method::new
  38: Returns a new instance for a given proxy. If a window (win) or a composite view is given, it uses this.
  39  

HelpSource/Classes/Object.schelp:
  114  
  115: method::rank
  116: Returns the number of dimensions the value has.
  117  A single value (scalar) has rank of zero.

HelpSource/Classes/Pdef.schelp:
  142  
  143: method::endless
  144: Returns a link::Classes/Prout:: that plays the proxy endlessly, replacing strong::nil:: with a strong::default:: value (silent event). This allows to create streams that idle on until a new pattern is inserted.
  145  

  186  
  187: method::isPlaying
  188: Returns true if player is running. If a Pdef is playing and its stream ends, it will schedule a stream for playing strong::as soon as a new one is assigned to it::. If it is stopped by strong::stop::, it won't.
  189  

HelpSource/Classes/Point.schelp:
  31  
  32: method::hash
  33: Returns a hash value for the receiver.
  34  

  80  
  81: method::asPoint
  82: Returns the receiver.
  83  
  84: method::asComplex
  85: Returns a complex number with x as the real part and y as the imaginary part.
  86  

HelpSource/Classes/PriorityQueue.schelp:
  15  
  16: method::topPriority
  17: Returns the time of the earliest item in the queue.
  18  
  19: method::pop
  20: Returns the earliest item in the queue.
  21  

HelpSource/Classes/Prout.schelp:
  13  
  14: method::new
  15: Returns a routine from the function.
  16  

HelpSource/Classes/PV_ChainUGen.schelp:
  18  
  19: method::fftSize
  20: Returns the FFT chain buffer's size.
  21  

HelpSource/Classes/Rect.schelp:
  59  
  60: method::moveBy
  61: Returns a new Rect which is offset by x and y.
  62  
  63: method::moveTo
  64: Returns a new Rect whose upper left corner is moved to (x, y).
  65  
  66: method::moveToPoint
  67: Returns a new Rect whose upper left corner is moved to aPoint.
  68  
  69: method::resizeBy
  70: Returns a new Rect whose dimensions have been changed by (x, y).
  71  
  72: method::resizeTo
  73: Returns a new Rect whose dimensions are (x, y).
  74  
  75: method::insetBy
  76: Returns a new Rect whose boundaries have been inset by (x, y). If only one argument is supplied, it will be used for both x and y.
  77  
  78: method::insetAll
  79: Returns a new Rect whose boundaries have been inset by the given amounts.
  80  

HelpSource/Classes/Scale.schelp:
   94  
   95: method::semitones
   96: Returns a tuned array of semitone values. link::#-as::(Array) is equivalent; link::#-as::(List) returns it as a list, etc.
   97  
   98: method::cents
   99: Returns a tuned array of cent values.
  100  
  101: method::ratios
  102: Returns a tuned array of ratios.
  103  

  129  
  130: method::size
  131: Returns the length of the scale.
  132  code::

  138  
  139: method::pitchesPerOctave
  140: Returns the size of the pitch class set from which the tuning is drawn.
  141  code::

  168  
  169: method::degreeToFreq
  170: Returns a frequency based on current tuning and rootFreq argument.
  171  code::

  175  
  176: method::degreeToRatio
  177: Returns a ratio based on current tuning.
  178  code::

HelpSource/Classes/Scheduler.schelp:
  38  
  39: method::isEmpty
  40: Returns whether the scheduling queue is empty.
  41  

HelpSource/Classes/SequenceableCollection.schelp:
  191  
  192: method::indexIn
  193: Returns the closest index of the value in the collection (collection must be sorted).
  194  code::

  197  
  198: method::indexInBetween
  199: Returns a linearly interpolated float index for the value (collection must be sorted). Inverse operation is link::#-blendAt::.
  200  code::

  204  
  205: method::blendAt
  206: Returns a linearly interpolated value between the two closest indices. Inverse operation is link::#-indexInBetween::.
  207  code::

  307  
  308: method::join
  309: Returns a link::Classes/String:: formed by connecting all the elements of the receiver, with strong::joiner:: inbetween. See also link::Classes/String#-split:: as the complementary operation.
  310  code::

  314  
  315: method::flat
  316: Returns a collection from which all nesting has been flattened.
  317  code::

  321  
  322: method::flatten
  323: Returns a collection from which strong::numLevels:: of nesting has been flattened.
  324  

  449  
  450: method::maxSizeAtDepth
  451: Returns the maximum size of all subarrays at a certain depth (dimension)
  452  

  464  
  465: method::maxDepth
  466: Returns the maximum depth of all subarrays.
  467  

  474  
  475: method::isSeries
  476: Returns true if the collection is an arithmetic series.
  477  

  490  
  491: method::resamp0
  492: Returns a new Collection of the desired length, with values resampled evenly-spaced from the receiver without interpolation.
  493  code::

  497  
  498: method::resamp1
  499: Returns a new Collection of the desired length, with values resampled evenly-spaced from the receiver with linear interpolation.
  500  code::

  584  
  585: method::integrate
  586: Returns a collection with the incremental sums of all elements.
  587  code::

  590  
  591: method::differentiate
  592: Returns a collection with the pairwise difference between all elements.
  593  code::

  623  
  624: method::convertDigits
  625: Returns an integer resulting from interpreting the elements as digits to a given base (default 10). See also asDigits in link::Classes/Integer#-asDigits:: for the complementary method.
  626  code::

  631  
  632: method::hammingDistance
  633: Returns the count of array elements that are not equal in identical positions. http://en.wikipedia.org/wiki/Hamming_distance
  634  

  644  
  645: method::editDistance
  646: Returns the minimum number of changes to modify this teletype::SequenceableCollection:: into the other teletype::SequenceableCollection::.
  647  A change can be: an addition, a deletion, or a substitution.

  693  
  694: method::similarity
  695: Returns a value between 0 and 1 representing the percentage similarity between this teletype::SequenceableCollection:: and the other teletype::SequenceableCollection::.
  696  

HelpSource/Classes/Set.schelp:
  31  
  32: method::includes
  33: Returns true if the specified item is present in the Set. Elements are checked for equality (not for identity).
  34  code::

  37  
  38: method::findMatch
  39: Returns the item, if it is present in the set. Otherwise returns nil. Element is checked for equality (not for identity).
  40  code::

  51  
  52: method::keyAt
  53: Returns the object at the internal strong::index::. This index is not deterministic.
  54  

  91  
  92: method::isSubsetOf
  93: Returns true if all elements of this are also elements of strong::that::. Elements are checked for equality (not for identity).
  94  Since Set is an unordered collection, order doesn't matter in this comparison.

HelpSource/Classes/SoundFile.schelp:
  81  
  82: method::collect
  83: Returns an link::Classes/Array:: of SoundFile objects whose paths match the pattern. (The associated files are closed. These objects can be used to cue playback buffers)
  84  code::

HelpSource/Classes/String.schelp:
   65  
   66: method::ascii
   67: Returns an Array of ASCII values of the Strings's characters.
   68  code::

   92  
   93: method::compare
   94: Returns an integer less than, equal to or greater than zero, depending on whether the receiver should be sorted before the argument, is equal to the argument or should be sorted after the argument. This is a case sensitive compare.
   95  

  223  
  224: method::asCompileString
  225: Returns a String formatted for compiling.
  226  code::

  465  
  466: method::find
  467: Returns the index of the string in the receiver, or nil if not found. If strong::ignoreCase:: is true, find makes no difference between uppercase and lowercase letters. The strong::offset:: is the point in the string where the search begins. string may be a String or a link::Classes/Char::.
  468  code::

  480  
  481: method::findAll
  482: Returns the indices of the string in the receiver, or nil if not found.
  483  code::

  487  
  488: method::contains
  489: Returns a link::Classes/Boolean:: indicating if the String contains string.
  490  code::

  500  
  501: method::containsStringAt
  502: Returns a link::Classes/Boolean:: indicating if the String contains string beginning at the specified index.
  503  code::

  510  method::beginsWith
  511: method::endsWith
  512: Returns true if this string begins/ends with the specified other string.
  513  argument:: string

  538  
  539: method::format
  540: Returns a formatted string with arguments. The % character in the format string is replaced by a string representation of an argument. To print a % character use \\% .
  541  code::

  604  
  605: method::stripRTF
  606: Returns a new String with all RTF formatting removed.
  607  code::

  615  
  616: method::split
  617: Returns an Array of Strings split at the separator. The separator is a link::Classes/Char::, and is strong::not:: included in the output array.
  618  code::

  720  
  721: method::getenv
  722: Returns the value contained in the environment variable indicated by the String.
  723  code::

  732  
  733: method::pathMatch
  734: Returns an link::Classes/Array:: containing all paths matching this String. Wildcards apply, non-recursive.
  735  code::

  888  
  889: method::findHelpFile
  890: Returns the path for the helpfile named this, if it exists, else returns nil.
  891  code::

  904  
  905: method::inspectorClass
  906: Returns class link::Classes/StringInspector::.
  907  

HelpSource/Classes/Tdef.schelp:
   98  
   99: method::endless
  100: Returns a link::Classes/Prout:: that plays the task endlessly, replacing strong::nil:: with a strong::default:: value 1. This allows to create streams that idle on until a new pattern is inserted.
  101  

  146  
  147: method::isPlaying
  148: Returns true if player is running. If a Tdef is playing and its stream ends, it will schedule a stream for playing strong::as soon as a new one is assigned to it::. If it is stopped by strong::stop::, it won't.
  149  

HelpSource/Classes/TempoClock.schelp:
  213  
  214: method::queue
  215: Returns the scheduling queue Array in the form [beat, function]. The maximum number of items is determined by the clock's queueSize argument upon instantiation. The default queueSize of 256 allows 128 functions to be in the queue at any time.
  216  
  217: method::beatDur
  218: Returns the duration in seconds of a current whole beat.
  219  

  260  
  261: method::bar
  262: Returns the current bar. See link::#-bars2beats:: for returning beat of current bar.
  263  
  264: method::nextBar
  265: Returns the number of beats at the next bar line relative to the beat argument. If strong::beat:: is not supplied, returns the beat at which the next bar begins.
  266  
  267: method::beatInBar
  268: Returns the current bar beat (as a link::Classes/Float::) in relation to link::#-beatsPerBar::. Values range from 0 to < beatsPerBar.
  269  
  270: method::baseBar
  271: Returns bar at which link::#-beatsPerBar:: was last changed. If beatsPerBar has not been changed since the clock was created, returns 0.
  272  
  273: method::baseBarBeat
  274: Returns beat at which the link::#-beatsPerBar:: was last changed. If beatsPerBar has not been changed since the clock was created, returns 0.
  275  
  276: method::beats2bars
  277: Returns a bar as a float relative to link::#-baseBarBeat::.
  278  
  279: method::bars2beats
  280: Returns a beat relative to link::#-baseBar::.
  281  code::

  285  
  286: method::timeToNextBeat
  287: Returns the logical time to next beat. strong::quant:: is 1 by default, relative to baseBarBeat, see link::#-nextTimeOnGrid::.
  288  

  298  
  299: method::elapsedBeats
  300: Returns the current elapsed time in beats. This is equivalent to code::tempoClock.secs2beats(Main.elapsedTime)::. It is often preferable to use link::#-beats:: instead of elapsedBeats because beats uses a thread's logical time.
  301  
  302: method::seconds
  303: Returns the current elapsed time. (This method is inherited from link::Classes/Clock::.)
  304  

HelpSource/Classes/Tuning.schelp:
  56  
  57: method::semitones
  58: Returns an array of semitone values for the pitch set. link::#-as::(Array) is equivalent; link::#-as::(List) returns it as a list, etc.
  59  
  60: method::cents
  61: Returns a array of cent values for the pitch set.
  62  
  63: method::ratios
  64: Returns a tuned array of ratios for the pitch set.
  65  

HelpSource/Classes/UnixFILE.schelp:
   9  
  10: method::isOpen
  11: Returns whether the file is open. An open request can fail if a file cannot be found for example. This method lets you test that the open call succeeded.
  12  

HelpSource/Overviews/GenericCollectors.schelp:
   8  
   9: method::add
  10: Returns an array with the value. This makes it unnecessary to initialize when adding to a variable.
  11  code::

  16  
  17: method::addAll
  18: Returns an array with all the values. This makes it unnecessary to initialize when adding to a variable.
  19  code::

  43  
  44: method::addFunc
  45: Returns a function or a FunctionList.
  46  This method is used to add multiple functions to already existing ones.

HelpSource/Overviews/Operators.schelp:
  126  
  127: method::rand
  128: Returns an evenly distributed random value between this and zero.
  129  code::

  134  
  135: method::rand2
  136: Returns an evenly distributed random value between [+this ... - this].
  137  code::

  142  
  143: method::linrand
  144: Returns a linearly distributed random value between this and zero.
  145  code::

  151  
  152: method::bilinrand
  153: Returns a linearly distributed random value between [+this ... - this].
  154  code::

  159  
  160: method::sum3rand
  161: Returns a value from a gaussian-like random distribution between this and zero.
  162  This was suggested by Larry Polansky as a loose approximation of gaussian. Follows the formula:

  173  
  174: method::coin
  175: Returns one or zero with the probability given by the argument.
  176  code::
```